frontend/src/components/AddDatatsetServiceAreaSelector/AddDatasetServiceAreaSelector.tsx

```diff
@@ -0,0 +1,44 @@
+import { MenuItem, SelectField } from 'material-ui'
+import * as React from 'react'
+import { DatasetCountySelection } from '../../constants/datatypes'
+import { State } from '../../constants/states'
+import { Store, withStore } from '../../services/store'
+import { StateSelector } from '../Selectors/StateSelector'
+
+export let AddDatasetServiceAreaSelector = withStore(
+    'selectedState',
+    'addDatasetCountySelection'
+)(({ store }) =>
+    <div>
+        <StateSelector
+            onChange={state => onStateChange(state, store)}
+            value={store.get('selectedState')}
+        />
+        <CountySelector
+            onChange={value => store.set('addDatasetCountySelection')(value)}
+            value={store.get('addDatasetCountySelection')}
+        />
+    </div>
+)
+
+function onStateChange(state: State, store: Store) {
+    store.set('selectedState')(state)
+    store.set('uploadedServiceAreasFilename')(null)
+}
+
+type Props = {
+    onChange(value: DatasetCountySelection): void
+    value: DatasetCountySelection
+}
+
+let CountySelector: React.StatelessComponent<Props> = ({ onChange, value }) =>
```

Comment:
definitely weird to have this in here and state selector in its own file. thoughts on whether this should be in its own file or if we should move state selector into this file.